### DISCUSSION TITLE
Same Vlan ID with different subnets. Same site. How do you enter them?

### DISCUSSION THREAD

**Discussion by slatevt** [netbox-community/netbox#11639] [Q&A] [answered]
Very new to Netbox and (of course) starting off with IPAM.  Trying to figure out how to add multiple Vlans with the same ID but different subnets. I've tried searching for the answer to this but I'm not finding it. (Maybe not searching with the correct search string?)

We route a few Vlans centrally and are available throughout multiple buildings. But many Vlans are routed at the top of rack switch at a buildings edge. These have duplicate Vlan ID's since the Vlan ID never leaves the building. How would you enter these into Netbox? 

To clarify:

The subnets are different, but the Vlan ID's are duplicates. I've tried adding VRF's and prefixes first, but when I try to add the Vlan I get an error about a duplicate Vlan ID:  "**_...VLAN with this Group and ID already exists._**" This is in a single location/site with many buildings.

Example:

Building 1 - Vlan 100 - subnet 192.168.1.0/24
Building 1 - Vlan 200 - subnet 172.16.1.0/24

Building 2 - Vlan 100 - subnet 192.168.2.0/24
Building 2 - Vlan 200 - subnet 172.16.2.0/24

Building 3 - Vlan 100 - subnet 192.168.3.0/24
Building 3 - Vlan 200 - subnet 172.16.3.0/24

etc....

I tried creating sperate VRF's and prefixes for each building, but I can't create the Vlans.

Can someone point to where in the documentation I look to find out how to do this. Am I missing something obvious?

**Comment by mtinberg** [comment, marked as answer]
It sounds like you've organized the VLANs into VLAN Groups, VLAN ID numbers in VLAN Groups are enforced to be unique, if those are Site-local VLANs as you describe then you'll want to define a Site for every building and then create the VLAN record associated with the Site, which will enforce uniqueness per-Site.  If you want to organize the building Sites into a Region or a Site Group you can do that and VLAN Groups can be associate with a scope such as Region or Site Group as well.

So VLANs can be unassociated and non-unique, or associated with a Site _or_ a VLAN Group with uniqeness.  And VLAN Groups can be associated with Rack, Location, Site, Region, Site Group, Cluster, and Cluster Group, effectively scoping the VLANs to those other objects and the Devices associated with them.  Associating a VLAN with a Site scopes the VLAN to the Site and Devices also associated with that Site.

Does this make more sense?

—

Mark Tinberg ***@***.***>

Division of Information Technology-Network Services

University of Wisconsin-Madison


________________________________
From: slatevt ***@***.***>
Sent: Wednesday, February 1, 2023 1:57 PM
To: netbox-community/netbox ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [netbox-community/netbox] Same Vlan ID with different subnets. Same site. How do you enter them? (Discussion #11639)


Very new to Netbox and (of course) starting off with IPAM. Trying to figure out how to add multiple Vlans with the same ID but different subnets. I've tried searching for the answer to this but I'm not finding it. (Maybe not searching with the correct search string?)

We route a few Vlans centrally and are available throughout multiple buildings. But many Vlans are routed at the top of rack switch at a buildings edge. These have duplicate Vlan ID's since the Vlan ID never leaves the building. How would you enter these into Netbox?

To clarify:

The subnets are different, but the Vlan ID's are duplicates. I've tried adding VRF's and prefixes first, but when I try to add the Vlan I get an error about a duplicate Vlan ID: "...VLAN with this Group and ID already exists." This is in a single location/site with many buildings.

Example:

Building 1 - Vlan 100 - subnet 192.168.1.0/24
Building 1 - Vlan 200 - subnet 172.16.1.0/24

Building 2 - Vlan 100 - subnet 192.168.2.0/24
Building 2 - Vlan 200 - subnet 172.16.2.0/24

Building 3 - Vlan 100 - subnet 192.168.3.0/24
Building 3 - Vlan 200 - subnet 172.16.3.0/24

etc....

I tried creating sperate VRF's and prefixes for each building, but I can't create the Vlans.

Can someone point to where in the documentation I look to find out how to do this. Am I missing something obvious?

**Comment by slatevt** [comment]
Yes... and thank you. This does make sense. 

It is making me rethink how I approach this.

We are still trying to determine if we want to move forward and use Netbox as our (at a minimum) ipam solution. So far its been a bit of work to get  information into it.  (I've been writing perl scripts to parse our IP data into csv files in order to import into Netbox.)

And slightly off topic here, do you happen to use Netbox for it's Rack feature? If so, do you find it complete enough to spend the time and effort to get all the necessary information into Netbox? Which, like the ipam piece, looks like a pretty daunting task. We have a lot of racks spread across a lot of buildings.

Again, thanks for the help.

**Comment by mtinberg** [comment]
> perl scripts to parse our IP data into csv files in order to import into Netbox

I have some perl tooling too for provisioning IP Address, Prefix, VLAN records from our internal home-grown IPAM database using the API, so that it can be kept up-to-date while we migrate the source-of-truth from our legacy system to Netbox.  Netbox is providing a read-only view of this data to other new tools, which can become read/write when we change which systems is the source.  CSV input can be more useful for ongoing migration now that it's possible to update records using CSV and not just create them one-time, as a lot of migrations end up needing to run both systems in parallel for an extended period of time.

> do you happen to use Netbox for it's Rack feature?

I'd like to, but Rack data entry can't be automated (there is no SNMP MIB that tells me what rack unit a device is in 😉) and our network engineers were using a mix of spreadsheets and OpenDCIM (which has a fancy JavaScript popup for placing devices in a rack that the engineers liked), so they haven't been motivated to use Netbox instead.  Knowing what building (Site) and room (Location) something is in can be enough, and we already encode that into the device name so staff can locate them easily, regardless of whether the Rack data is populated.

I'd show people how to use the Rack maps and see if they are motivated and find value in keeping the data up to date, and if they don't see the value then the rest of Netbox can still be useful.  Rack mapping is a manual process that is separable from the rest of Netbox as it often doesn't change the device config in any way (unless you have VLANs assigned to Racks or something), and I see Netbox as primarily a system for storing the data for config templates and config automation.


—

Mark Tinberg ***@***.***>

Division of Information Technology-Network Services

University of Wisconsin-Madison


________________________________
